src/utils/circular-queue-provider.ts

```diff
@@ -0,0 +1,96 @@
+export class Observer {
```

Comment:
Maybe for this class should be created another file, to keep the same structure.